src/client/dfs/dfs_internal.h

```diff
@@ -53,6 +53,22 @@ dfs_lookupx(dfs_t *dfs, dfs_obj_t *parent, const char *name, int flags,
 	    dfs_obj_t **obj, mode_t *mode, struct stat *stbuf, int xnr,
 	    char *xnames[], void *xvals[], daos_size_t *xsizes);
 
+/* oid-in variants of dfs functions, these can be be used to verify the file being clobbered */
+int
+dfs_removeoi(dfs_t *dfs, dfs_obj_t *parent, const char *name, bool force, daos_obj_id_t *oid);
+
+/* moid is moved oid, oid is clobbered file.
+ * This isn't yet fully compatible with dfuse because we also want to pass in a flag for if the
+ * destination exists.
+ */
+int
+dfs_moveoi(dfs_t *dfs, dfs_obj_t *parent, char *name, dfs_obj_t *new_parent, char *new_name,
+	   daos_obj_id_t *moid, daos_obj_id_t *oid);
+
+/* Set the parent, but takes the parent, rather than another file object */
+void
+dfs_update_parentfd(dfs_obj_t *obj, dfs_obj_t *src_obj, const char *name);
```

Comment:
why is the new parent called src_obj? should call it new_parent

Reply:
agreed, this came straight from dfs_update_parent()